mh/session.py

```diff
@@ -73,17 +73,81 @@ def __init__(self, connection_handler):
+        }
+        return pdict
+
+    @staticmethod
+    def deserialize(pdict):
+        session = Session(None)
+        session.pat_ticket = str(pdict["pat_ticket"]) if pdict["pat_ticket"] else pdict["pat_ticket"]
+        session.local_info["server_id"] = int(pdict["local_info_server_id"]) if pdict["local_info_server_id"] else pdict["local_info_server_id"]
+        session.local_info["server_name"] = str(pdict["local_info_server_name"]) if pdict["local_info_server_name"] else pdict["local_info_server_name"]
+        session.local_info["gate_id"] = int(pdict["local_info_gate_id"]) if pdict["local_info_gate_id"] else pdict["local_info_gate_id"]
+        session.local_info["gate_name"] = str(pdict["local_info_gate_name"]) if pdict["local_info_gate_name"] else pdict["local_info_gate_name"]
+        session.local_info["city_id"] = int(pdict["local_info_city_id"]) if pdict["local_info_city_id"] else pdict["local_info_city_id"]
+        session.local_info["city_name"] = str(pdict["local_info_city_name"]) if pdict["local_info_city_name"] else pdict["local_info_city_name"]
+        session.local_info["city_size"] = int(pdict["local_info_city_size"]) if pdict["local_info_city_size"] else pdict["local_info_city_size"]
+        session.local_info["city_capacity"] = int(pdict["local_info_city_capacity"]) if pdict["local_info_city_capacity"] else pdict["local_info_city_capacity"]
+        session.local_info["circle_id"] = int(pdict["local_info_circle_id"]) if pdict["local_info_circle_id"] else pdict["local_info_circle_id"]
+        session.online_support_code = str(pdict["online_support_code"]) if pdict["online_support_code"] else pdict["online_support_code"]
+        session.capcom_id = str(pdict["capcom_id"])
+        session.hunter_name = str(pdict["hunter_name"])
+        session.hunter_stats = pdict["hunter_stats"]
+        session.layer = int(pdict["layer"])
+        session.state = int(pdict["state"])
+        session.binary_setting = pdict["binary_setting"]
+        h_settings = bytearray(pdict["hunter_info"], encoding='ISO-8859-1')
+        session.hunter_info = pati.HunterSettings().unpack(h_settings, len(h_settings))
+        return session
+
+    def get(self, connection_data, wait_for_session=False):
         """Return the session associated with the connection data, if any."""
-        if hasattr(connection_data, "pat_ticket"):
-            self.pat_ticket = to_str(
-                pati.unpack_binary(connection_data.pat_ticket)
-            )
+        if wait_for_session:
+            if hasattr(connection_data, "pat_ticket"):
+                self.pat_ticket = to_str(
+                    pati.unpack_binary(connection_data.pat_ticket)
+                )
+                session = None
+                while session is None:
+                    try:
+                        session = STATE.get_session(self.pat_ticket)
+                    except Exception as _:
+                        pass
```

Comment:
Isn't there a cleaner way rather than shadowing the `Exception`?

Reply:
Corrected this issue.